src/apply_manual_codes.py

```diff
@@ -0,0 +1,156 @@
+        # Merge manually coded radio show files into the cleaned dataset
+        for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            rqa_messages = [td for td in data if plan.raw_field in td]
+            coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+    
+            f = None
+            try:
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                    user, rqa_messages, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+                
+                if plan.binary_code_scheme is not None:
+                    if f is not None:
+                        f.seek(0)
+                    TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
+                        user, rqa_messages, plan.id_field, {plan.binary_coded_field: plan.binary_code_scheme}, f)
+            finally:
+                if f is not None:
+                    f.close()
+
+        # Merge manually coded follow up surveys into the cleaned dataset
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            follow_up_messages = [td for td in data if plan.raw_field in td]
+            coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+    
+            f = None
+            try:
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                    user, follow_up_messages, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+            
+            finally:
+                if f is not None:
+                    f.close()
+        
+        # Label the RQA & surveys for which there is no response yet as TRUE MISSING
+        for td in data:
+            missing_dict = dict()
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+                if plan.raw_field not in td:
+                    na_label = CleaningUtils.make_label_from_cleaner_code(
+                        plan.code_scheme, plan.code_scheme.get_code_with_control_code(Codes.TRUE_MISSING),
+                        Metadata.get_call_location()
+                    )
+                    missing_dict[plan.coded_field] = [na_label.to_dict()]
+
+                    if plan.binary_code_scheme is not None:
+                        na_label = CleaningUtils.make_label_from_cleaner_code(
+                            plan.binary_code_scheme, plan.binary_code_scheme.get_code_with_control_code(Codes.TRUE_MISSING),
+                            Metadata.get_call_location()
+                        )
+                        missing_dict[plan.binary_coded_field] = na_label.to_dict()
+
+            td.append_data(missing_dict, Metadata(user, Metadata.get_call_location(), time.time()))
+            
```

Comment:
I've realised you do need another block of code in here that sets 'NC's on the RQAs and Follow-Ups where the raw text is "". You should be able to adapt the code you used for doing this with the demogs below.

Reply:
Updated PTAL @as2388

Reply:
You need to set the binary code scheme to NC too